src/endpoint/s3/ops/s3_put_bucket_lifecycle.js

```diff
@@ -132,11 +140,18 @@ async function put_bucket_lifecycle(req) {
         }
 
         if (rule.Expiration?.length === 1) {
+            if (rule.Expiration.length > 1) {
+                throw new S3Error(S3Error.MalformedXML);
```

Comment:
maybe we should have a more verbose message of what went wrong? is this what AWS returns?

Reply:
yes aws returns XML error for this case. Added message, thanks